crates/solver/src/driver.rs

```diff
@@ -283,52 +280,25 @@ impl Driver {
             }
         });
 
-        let previous_orders: Vec<OrderUid> = auction
-            .orders
-            .iter()
-            .map(|order| order.metadata.uid)
-            .collect();
         let balance_start = Instant::now();
-        let orders = self.order_balance_filter.filter(auction.orders).await;
+        let balances =
+            order_balance_filter::fetch_balances(self.balance_fetcher.as_ref(), &auction.orders)
+                .await;
         tracing::debug!(
-            "filtering orders based on balance took {}s",
+            "fetching order balances took {}s",
```

Comment:
I am a bit worried this will take too long in production. We need to fetch something around 1000 balances at the moment. Can you run this locally connected to the production orderbook to verify that there is no issue with this taking too long?

I think this is something that came up before, but I can't remember the reasoning behind having the `solver` re-fetch the balances instead of including them as part of each order in an auction (since the autopilot already fetches these are part of its auction updating loop).

Reply:
This is not new in this PR. It already runs in production. It came up in the PR that introduced it [here](https://github.com/cowprotocol/services/pull/1434#pullrequestreview-1391989453) . According to the log message it always takes less than 1 sec i prod.

Reply:
One thing that I just realizes complicates things, is that balance fetching now has to take pre- and post- interactions into account in both the driver and the autopilot. Not a huge deal, but something that is important to remember.